Source/SwiftLintFramework/Rules/Lint/DynamicInlineRule.swift

```diff
@@ -25,47 +24,33 @@ public struct DynamicInlineRule: ASTRule, ConfigurationProviderRule {
         ]
     )
 
-    public func validate(file: SwiftLintFile, kind: SwiftDeclarationKind,
-                         dictionary: SourceKittenDictionary) -> [StyleViolation] {
-        // Look for functions with both "inline" and "dynamic". For each of these, we can get offset
-        // of the "func" keyword. We can assume that the nearest "@inline" before this offset is
-        // the attribute we are interested in.
-        guard functionKinds.contains(kind),
-            case let attributes = dictionary.enclosedSwiftAttributes,
-            attributes.contains(.dynamic),
-            attributes.contains(.inline),
-            let funcOffset = dictionary.offset.flatMap(file.stringView.location),
-            case let inlinePattern = regex("@inline"),
-            case let range = NSRange(location: 0, length: funcOffset),
-            let inlineMatch = inlinePattern.matches(in: file.contents, options: [], range: range)
-                .last,
-            inlineMatch.range.location != NSNotFound,
-            case let attributeRange = NSRange(location: inlineMatch.range.location,
-                                              length: funcOffset - inlineMatch.range.location),
-            case let alwaysInlinePattern = regex("@inline\\(\\s*__always\\s*\\)"),
-            alwaysInlinePattern.firstMatch(in: file.contents, options: [], range: attributeRange) != nil
-        else {
-            return []
+    public func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor? {
+        Visitor()
+    }
+}
+
+private extension DynamicInlineRule {
+    private final class Visitor: SyntaxVisitor, ViolationsSyntaxVisitor {
+        private(set) var violationPositions: [AbsolutePosition] = []
+
+        override func visitPost(_ node: FunctionDeclSyntax) {
```

Comment:
This won't cover subscripts, inits and getter/setters, which the rule currently covers (based on `functionKinds`, not sure if that actually works/is expected)

Reply:
Good catch! The example/test coverage for this rule is not great so it didn't catch any of that.

Reply:
I merged this by mistake without addressing your feedback. I'll be doing that in a new PR.